content-scripts/components/Button.tsx

```diff
@@ -0,0 +1,20 @@
+import jsx from "texsaur";
+
+interface ButtonProps {
+    title?: string;
+    icon?: string;
+    id?: string;
+    className?: string;
+    onClick?: (e: Event) => void;
+}
+
+const Button: JSX.Component<ButtonProps> = ({ title, icon, id, className, onClick }) => {
+    return (
+        <button id={id} className={className} onClick={onClick}>
+            <span className={"se-icon " + icon}></span>
+            {title}
```

Comment:
I'm not sure if this would be ideal but there might be some cases where the content of the button is more than an icon and some text, in those cases we maybe should leave the onus of the content to the "parent" of the Button (as in wherever it is being called). 
If there is no `icon` defined this implementation will still place a `<span>` unnecessarily. We'll also want to account for multiple types of buttons (primary, secondary, ...) these may need to have some additional properties such as disabled. 
Also what's your opinion on having a "type" prop and handling the styling based on it?  And should we have anchor tags also be Buttons, there are many instances where we want to make a button that is simply an `<a href=.../>`?

Reply:
I am thinking this as atom that should be the last child in the tree.
I like the types and think it is a good idea. That should be an issue to apply our own design later

Reply:
That would make sense, we could leave this be as is for now and if the problem arises in the future we'll handle it then. I agree, adding the other stylings/types should be done in another PR.